dtree.c

```diff
@@ -347,6 +361,59 @@ int getHeight (char* message, bool wrap){
 }
 
 
+char* getEndOfLine(char* line_start, int wrap){
+    if ( !*line_start ) return NULL;
```

Comment:
Unfortunately this isn't the case. Without going into too much detail, the case you refer to in the loop returns a valid pointer, where this returns null. We specifically need the case where if the address pointed to by line_start is null, we return a null value, otherwise this code will loop infinitely.

Reply:
Probably should add another case for `line_start` being NULL, because this is dangerous.